internal/temporal/workflows/export/export_workflow.go

```diff
@@ -0,0 +1,262 @@
+	}
+
+	logger.Info("Using execution time for interval boundaries", "execution_time", executionTime)
+
+	var taskDetails export.ScheduledTaskDetails
+	err := workflow.ExecuteActivity(ctx, scheduledTaskActivity.GetScheduledTaskDetails, export.GetScheduledTaskDetailsInput{
+		ScheduledTaskID:       input.ScheduledTaskID,
+		TenantID:              input.TenantID,
+		EnvID:                 input.EnvID,
+		WorkflowExecutionTime: executionTime,         // Pass scheduled execution time
+		CustomStartTime:       input.CustomStartTime, // Pass custom time range for force runs
+		CustomEndTime:         input.CustomEndTime,
+	}).Get(ctx, &taskDetails)
+	if err != nil {
+		logger.Error("Failed to get scheduled task details", "error", err)
+		return nil, fmt.Errorf("failed to get scheduled task: %w", err)
+	}
+
+	// Check if task is enabled
+	if !taskDetails.Enabled {
+		logger.Info("Scheduled task is disabled, skipping execution")
+		return &ExecuteExportWorkflowOutput{
+			Success: true,
+		}, nil
+	}
+
+	logger.Info("Scheduled task details retrieved",
+		"entity_type", taskDetails.EntityType,
+		"start_time", taskDetails.StartTime,
+		"end_time", taskDetails.EndTime)
+
+	// ================================================================================
+	// STEP 2: Generate Task ID
+	// ================================================================================
+	logger.Info("Step 2: Generating task ID")
+	var taskID string
+	err = workflow.SideEffect(ctx, func(ctx workflow.Context) interface{} {
+		return types.GenerateUUIDWithPrefix(types.UUID_PREFIX_TASK)
+	}).Get(&taskID)
+	if err != nil {
+		logger.Error("Failed to generate task ID", "error", err)
+		return nil, fmt.Errorf("failed to generate task ID: %w", err)
+	}
+
+	workflowID := workflow.GetInfo(ctx).WorkflowExecution.ID
+	logger.Info("Generated task ID", "task_id", taskID, "workflow_id", workflowID)
+
+	// ================================================================================
+	// STEP 3: Create Task Record
+	// ================================================================================
+	logger.Info("Step 3: Creating task in database")
+	createTaskInput := export.CreateTaskInput{
+		TaskID:          taskID,
+		WorkflowID:      workflowID,
+		ScheduledTaskID: input.ScheduledTaskID,
+		TenantID:        taskDetails.TenantID,
+		EnvID:           taskDetails.EnvID,
+		UserID:          input.UserID, // Use user ID from input (empty for scheduled runs)
+		EntityType:      taskDetails.EntityType,
+		StartTime:       taskDetails.StartTime,
+		EndTime:         taskDetails.EndTime,
+	}
+
+	var createTaskOutput export.CreateTaskOutput
+	err = workflow.ExecuteActivity(ctx, taskActivity.CreateTask, createTaskInput).Get(ctx, &createTaskOutput)
+	if err != nil {
+		logger.Error("Failed to create task", "error", err)
+		return nil, fmt.Errorf("failed to create task: %w", err)
+	}
+
+	logger.Info("Task created in database", "task_id", taskID)
```

Comment:
lets try this once: fail the task at step 3 and retry workflow after this breaks